doc/how-to-contribute.rst

```diff
@@ -56,3 +56,94 @@ favourite tool easily.
 .. _Sphinx: http://www.sphinx-doc.org/en/stable
 .. _PyLint: https://www.pylint.org
 .. _Codacy: https://www.codacy.com/app/tango-controls/pytango/dashboard
+
+Releasing a new version
+-----------------------
+
+From time to time a new version should be released.  The basic steps required are
+as follows:
+
+Pick a version number
+  * Semantic version numbering is used:  <major>.<minor>.<patch>
+  * The major and minor version fields (9.2) track the TANGO C++ core version.
+  * For small changes, increase the patch field (last number), for example:
+    9.2.4 -> 9.2.5
+
+Create an issue in Github
+  * This is to inform the community that a release is planned.
+  * Use a checklist similar to the one below:
+
+    | Task list:
+    | - [ ] Read steps in the how-to-contribute docs for making a release
+    | - [ ] Edit the changelog
+    | - [ ] Bump the version
+    | - [ ] Merge develop into stable
+    | - [ ] Make sure Travis is OK on stable branch
+    | - [ ] Make sure the documentation is updated (readthedocs)
+    | - [ ] Create a release tag on GitHub, from stable branch
+    | - [ ] Upload the new version to PyPI
+    | - [ ] Bump the version with "-dev" in the develop branch
+    | - [ ] Fill the release description on GitHub
+    | - [ ] Build conda packages
+    | - [ ] Advertise the release on the mailing list
+    | - [ ] Close this issue
+
+  * A check list is this form on github can be ticked off as the work progresses.
+
+Make a branch from `develop` to prepare the release
+  * Example branch name: ``prepare-v9.2.5``.
+  * Edit the changelog (in ``docs/revision.rst``).  Include *all* pull requests
+    since the previous release.
+  * Bump the versions in (in ``tango/release.py``).  E.g. ``version_info`` to (9, 2, 5)
+  * Create a pull request to get these changes reviewed before proceeding.
+
+Merging ``develop`` into ``stable``
+  * Wait until the preparation branch pull request has been merged.
+  * Merge ``stable`` into the latest ``develop``, and push to github.
+  * In general, the ``stable`` branch should point to the latest release.
```

Comment:
> Merge ``stable`` into the latest ``develop``

It should be the opposite right? Also I'd recommend a fast-forward merge in order to avoid a confusing merge commit. This can also be done by simply pushing develop to stable: 

    git push origin develop:stable

This way the release tag corresponds to the actual release commit both on the stable and develop branches.